core/src/main/java/org/radargun/stages/cache/background/PrivateLogLogic.java

```diff
@@ -48,12 +51,13 @@ protected boolean invokeLogic(long keyId) throws Exception {
          if (prevValue == null || !prevValue.contains(prevOperation.operationId)) {
             // non-cleaned old value or stale read, try backup
             backupValue = checkedGetValue(~keyId);
-            if (backupValue == null || !backupValue.contains(prevOperation.operationId)) {
+            // Modifying the same key within a single transaction may cause false stale reads, avoid it by checking txModificationKeyIds
```

Comment:
As far as I remember the was related to reaching max size & triming checked operations from value. I was hitting the issue regularly, therefore I would prefer to keep current solution, which doesn't spoil test runs. I can add TODO so that we make sure the problem can be handled inthe future when we have better test coverage.